runtime/amd64.S

```diff
@@ -1189,20 +1189,22 @@ ENDFUNCTION(G(caml_reperform))
 
 FUNCTION(G(caml_resume))
 CFI_STARTPROC
-    /* %rax -> fiber, %rbx -> fun, %rdi -> arg */
+    /* %rax -> fiber, %rbx -> fun, %rdi -> arg, %rsi -> last_fiber */
         leaq    -1(%rax), %r10  /* %r10 (new stack) = Ptr_val(%rax) */
-        movq    %rdi, %rax      /* %rax := argument to function in %rbx */
+        movq    %rdi, %rax      /* %rax := argument to function in %rdi */
     /*  check if stack null, then already used */
         testq   %r10, %r10
-        jz      2f
+        jz      1f
 #if defined(WITH_THREAD_SANITIZER)
-    /* Save non-callee-saved registers %rax and %r10 before C call */
+    /* Save non-callee-saved registers %rax, %rsi and %r10 before C call */
         ENTER_FUNCTION
         pushq   %rax; CFI_ADJUST(8);
+        pushq   %rsi; CFI_ADJUST(8);
         pushq   %r10; CFI_ADJUST(8);
     /* Necessary to include the caller of caml_resume in the TSan backtrace */
         TSAN_ENTER_FUNCTION(16)
```

Comment:
This now needs to be
```suggestion
        TSAN_ENTER_FUNCTION(24)
```
because you've saved one more register to the stack.
Without this change, `tests/tsan/perform.ml` and `tests/tsan/reperform.ml` fail for tsan-enabled compilers.